Assets/MRTK/Providers/Experimental/WindowsSceneUnderstanding/WindowsSceneUnderstandingObserver.cs

```diff
@@ -660,7 +666,7 @@ private async Task RunObserverAsync(CancellationToken cancellationToken)
 
                     case ObserverState.GetScene:
                         observerState = ObserverState.Working;
-                        if (CreateGameObjects && instantiationQueue.Count > 0)
+                        while (CreateGameObjects && instantiationQueue.Count > 0)
```

Comment:
It's not clear to me that anything is changing with `instantiationQueue` in this `while`, so will this spin forever?

Reply:
This statement is in the `task` we run in the background and in `Update` we dequeue this queue based on the batch size we set. So this will not spin forever unless someone set the batch size to zero, which probably nobody should do. The original implementation only wait for one frame but that could lead to problems when the batch size is small so the execution continues before all queue gameobjects are instantiated.